### ISSUE TITLE
Change the way we pmap inside the pipeline

### ISSUE THREAD

**Issue by ufuk-cakir** [AstroAI-Lab/rubix#65]
Currently, I pmap only specific functions inside the pipeline, for example here:
https://github.com/ufuk-cakir/rubix/blob/a6da9e397a2f6ec584a02b0ded3bc571d5d7e13e/rubix/core/ssp.py#L60

When the pipeline is assembled, `jax.jit` is applied to the resulting concatenated function. This results in the warning from Jax:
```shell
UserWarning: The jitted function <unnamed wrapped function> includes a pmap. Using jit-of-pmap can lead to inefficient data movement, as the outer jit does not preserve sharded data representations and instead collects input and output arrays onto a single device. Consider removing the outer jit unless you know what you're doing. See

https://github.com/google/jax/issues/2926
``` 

# Solution
Instead of pmaping subfunction, define the pipeline to calculate one IFU cube on some input data, and then pmap this function such that each GPU works on a subset of the data, calculates the IFU cube, and in the end we sum the result up, to get the final cube.

This has the additional advantage, that even if the GPU memory is not enough, we can further split the data on two or more pipeline calls, and sum up the results in the end. 

## Caveats
We have to apply specific functions like psf, lsf and noise only on the final ifu cube after summation. Also we have to be careful if this breaks some physics assumptions
